src/pages/docs/guides/pub-sub/data-streaming.mdx

```diff
@@ -0,0 +1,545 @@
+
+[//]: # (Again, might be good to get a diagram or somethign in here so it feels less like a wall of text..)
+
+Server-side batching is best in scenarios like:
+- **Social feeds and reactions:** Likes, emoji reactions, comments during live events
+- **Chat applications:** High-activity chat rooms during key moments
+- **Event streams:** Real-time activity feeds with bursty traffic patterns
+
+### Configuration
+
+Configure server-side batching through [channel rules](/docs/channels#rules):
+
+1. Navigate to your app settings
+2. Under channel rules, create a new rule
+3. Specify the channel name or namespace pattern
+4. Enable server-side batching
+5. Set the batching interval (e.g., 100ms, 500ms, 1000ms)
+
+The batching interval determines the maximum delay before messages are delivered. Shorter intervals maintain lower latency but provide less message reduction. Longer intervals maximize cost savings but increase delivery delay.
+
+Each batch can contain up to 200 messages by count or data size. If more than 200 messages are published in a window, they're split into multiple batches automatically.
+
+**Important considerations:**
+- Server-side batching is mutually exclusive with [message conflation](/docs/messages#conflation)
+- Messages with explicit IDs (for [idempotency](/docs/pub-sub/advanced#idempotency)) are excluded from batching
+
+See [configure server-side batching](/docs/messages/batch#configure) for complete setup instructions.
+
+### Key considerations
+
+Before implementing server-side batching, consider the following:
+
+* **Latency impact:** Messages are delayed by the batching interval. A 100ms interval means 0-100ms additional delay per message. Can you applications tolerate this?
+* **Burst characteristics:** Batching is most effective during traffic spikes. Measure your actual burst patterns to choose optimal intervals. Steady, low-rate traffic may not benefit significantly.
+* **Batch size limits:** Each batch is limited to 200 messages or maximum data size. Higher rates may generate multiple batches per interval.
+* **Idempotency trade-off:** Messages with explicit IDs (for idempotency) are excluded from batching. If you need idempotent publishes, you cannot use server-side batching on those messages.
+* **Monitoring and alerting:** Track actual batch sizes and frequencies in production. Unexpectedly small batches may indicate misconfiguration or changing traffic patterns.
+* **Consumer processing:** Your clients should be able to handle bursts of messages arriving together. Consider client-side queuing or throttling if necessary.
+* **Mutual exclusivity with conflation:** You must choose between batching (deliver all messages) or conflation (deliver only latest). Plan channel namespaces accordingly if you need both patterns.
+
+### Implementation
+
+Server-side batching requires no code changes. Producers publish normally, and consumers receive batched messages transparently:
+
+<Code>
```

Comment:
Ditto re more languages - this one would probably benefit from more client-side languages so Java + Swift + Readt